### ISSUE TITLE
Add an option to distinguish exported types which are used in the same module vs. completely unused

### ISSUE THREAD

**Issue by danvk** [webpro/knip#149]
Say I define a tagged union in TypeScript:

```ts
export interface ThingOne { type: 'one' }
export interface ThingTwo { type: 'two' }
export type Thing = ThingOne | ThingTwo;
```

If only `Thing` and `ThingOne` are imported from other files, then knip would like me to rewrite this as:

```ts
export interface ThingOne { type: 'one' }
interface ThingTwo { type: 'two' }
export type Thing = ThingOne | ThingTwo;
```

I don't particularly like this from a symmetry perspective, and it implies a distinction between `ThingOne` and `ThingTwo` that I don't want to make (even when it's not exported, `ThingTwo` can be reconstructed as `Exclude<Thing, ThingOne>`).

I could just disable the "Unused export types" / `types` issue type, but I _do_ want to catch exported types that are completely unused (i.e. never imported and not used in their defining module):

```ts
export type CompletelyUnused = number;
```

These cases seem quite different to me and it would be nice if there were a way to distinguish them with knip. As a reference, I added a similar feature to ts-prune a few years back: https://github.com/nadeesha/ts-prune/pull/74

**Comment by webpro** [comment]
This feature already exists, but I realize it isn't documented yet 🤦 

For now, you can look for `ignoreExportsUsedInFile` (https://github.com/search?q=repo%3Awebpro%2Fknip%20ignoreExportsUsedInFile&type=code). It was introduced by @JoshuaKGoldberg in #139.

**Comment by danvk** [comment]
Thanks @webpro and @JoshuaKGoldberg ! Some mention in the docs would be nice, I actually just discovered it myself through TS autocomplete :)

<img width="608" alt="image" src="https://github.com/webpro/knip/assets/98301/7c80275a-6946-4717-942d-758c8d02d265">

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.15.0. See [Release 2.15.0](https://github.com/webpro/knip/releases/tag/2.15.0) for release notes._
